packages/fxa-settings/src/lib/channels/firefox.ts

```diff
@@ -78,12 +87,88 @@ export type FxAOAuthLogin = {
   state: string;
 };
 
+interface WebChannelRequest {
+  command: FirefoxCommand;
+  messageId: string;
+  timeoutId?: number;
+  reject(reason?: any): void;
+  resolve(value?: any): void;
+}
+
+interface OutstandingRequestOptions {
+  window: Window;
+  sendTimeoutLength?: number;
+}
+
+/**
+ * OutstandingRequests manages all Firefox WebChannel requests that
```

Comment:
Thanks! 🙌

While we're at it, do you have an opinion on what the web content should do if it never receives the web channel message back with CWTS engines? In Backbone, we seem to basically bork the entire screen with a cryptic message about `_getSyncEngines` failing. Right now in the React version, we just show a loading spinner. I thought about adding a timeout there where at least Desktop could show _some_ sync options, but left it since at least the user could still submit the form which was an improvement.

Reply:
Aaah good question - not sure what an ideal experience would be but sounds like what you did is already an improvement!

Having a clear error after some time and asking the user to reload the page might be OK